Monolith.Unity/GameBootProfile.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text;
+using Monolith.Extensions;
+using UnityEngine;
+
+namespace Monolith.Unity
+{
+    
+    [Serializable]
+    public sealed class GameBootProfile : IGameBootProfile
+    {
+
+        [SerializeField] private string _id;
+        [SerializeField] private GameBootOptions bootOptions;
+        [SerializeField] private string[] _scriptingDefineSymbols;
+        
+        public GameBootProfile()
+        {
+            _id = "profile";
+            bootOptions = null;
+            _scriptingDefineSymbols = new string[0];
+        }
+        
+        public string Id => _id ?? string.Empty;
+        public IGameBootOptions BootOptions => bootOptions;
+        public ReadOnlyCollection<string> ScriptingDefineSymbols => new ReadOnlyCollection<string>(_scriptingDefineSymbols ?? new string[0]);
+        
+        public bool NeedsSanitizing
+        {
+            get
+            {
+                return (
+                    string.IsNullOrEmpty(_id) ||
+                    (_scriptingDefineSymbols == null) ||
+                    _scriptingDefineSymbols.Any((symbol) => !symbol.IsDefineDirectiveSafe()) ||
+                    (_scriptingDefineSymbols.Distinct().Count() < _scriptingDefineSymbols.Length)
+                );
+            }
+        }
+
+        public void Sanitize()
+        {
+            if (Application.isPlaying) throw new InvalidOperationException();
+            
+            _id ??= string.Empty;
+            _scriptingDefineSymbols ??= new string[0];
+
+            int size = _scriptingDefineSymbols.Any() ? _scriptingDefineSymbols.Max((symbol) => symbol.Length) : 0;
+
+            var stringBuilder = new StringBuilder(size);
+            var list = new List<string>(_scriptingDefineSymbols.Length);
```

Comment:
Not a huge fan of var but you use it really well in these contexts!

Reply:
I don't like it when people overuse it either. I personally only use it when:
- The right hand side makes the type really explicit, e.g `new T()`, `GetT()`.
- The full type name is way too long. This can happen for many reasons: generics, tuples, complex LINQ expressions, wordy class names...